spark/spark-tensorflow-connector/src/main/scala/org/tensorflow/spark/datasources/tfrecords/serde/DefaultTfRecordRowEncoder.scala

```diff
@@ -111,9 +124,20 @@ object DefaultTfRecordRowEncoder extends TfRecordRowEncoder {
         FloatListFeatureEncoder.encode(ArrayData.toArrayData(row.get(index)).toFloatArray())
       case ArrayType(DoubleType, _) =>
         FloatListFeatureEncoder.encode(ArrayData.toArrayData(row.get(index)).toDoubleArray().map(_.toFloat))
+      case ArrayType(DecimalType(), _) =>
+        val decimalArray = ArrayData.toArrayData(row.get(index)).toArray[Decimal](DataTypes.createDecimalType())
+        FloatListFeatureEncoder.encode(decimalArray.map(_.toFloat))
       case ArrayType(_, _) =>
         BytesListFeatureEncoder.encode(ArrayData.toArrayData(row.get(index)).toArray[String](StringType))
-      case _ => BytesListFeatureEncoder.encode(Seq(row.getString(index)))
+      case VectorType => {
+        val field = row.get(index)
+        field match {
+          case v: SparseVector => FloatListFeatureEncoder.encode(v.toDense.toArray.map(_.toFloat))
```

Comment:
@pilipolio The tf.VarLenFeature is only available in the TensorFlow Python API. We could use the TensorFlow java API, but it was not available when we initially created the library and would involve a major re-write which we can address in a different PR.